packages/buidler-core/src/internal/buidler-evm/provider/provider.ts

```diff
@@ -215,54 +213,29 @@ export class BuidlerEVMProvider extends EventEmitter
       return;
     }
 
-    let compilerInput: CompilerInput | undefined;
-    let compilerOutput: CompilerOutput | undefined;
+    const buildInfos: BuildInfo[] = [];
 
-    if (this._solcVersion !== undefined && this._paths !== undefined) {
-      if (semver.lt(this._solcVersion, FIRST_SOLC_VERSION_SUPPORTED)) {
+    if (this._paths !== undefined) {
+      const buildInfoFiles = await getBuildInfoFiles(this._paths.artifacts);
+
+      try {
+        for (const buildInfoFile of buildInfoFiles) {
+          const buildInfo = await fsExtra.readJson(buildInfoFile);
+          if (semver.gte(buildInfo.solcVersion, FIRST_SOLC_VERSION_SUPPORTED)) {
```

Comment:
The previous version was showing a warning here, when some version didn't match. Should we also show something in this case? Maybe a warning that's only shown once saying something like "some of your contracts are using a solc older than 0.5.1 and won't show up in the node" (or something like that)